crates/orchestrator/src/workers/data_submission.rs

```diff
@@ -0,0 +1,48 @@
+use crate::config::config;
+use crate::jobs::create_job;
+use crate::jobs::types::{JobStatus, JobType};
+use crate::workers::Worker;
+use async_trait::async_trait;
+use std::collections::HashMap;
+use std::error::Error;
+
+pub struct DataSubmissionWorker;
+
+#[async_trait]
+impl Worker for DataSubmissionWorker {
```

Comment:
name is not very clear, afaiu this job's role is to make sure DA picks up proving right ? Do we never see a world where proving is ahead of DA ?

Reply:
yes but let's say DA jobs have failed then in that case proving is ahead ?
I think `DataAvailabilitySynchronizer` is a better name?

Reply:
@apoorvsadana ,
Note: name changed to `DataAvailabilitySynchronizer`.

Reply:
> yes but let's say DA jobs have failed then in that case proving is ahead ?

So I guess there's some confusion. Proving is always ahead. Only after a proving job is succesful we actually create a new job for DA. I would still like to keep the Worker name actually because its coherent with all other workers. wdyt

Reply:
yes confusion cleared, agreed lets keep it that way for now although it does not submit only for one block but its role is to keep DA and proving synchronized